File: Stack/Stack.java
// package Stack;

public class Stack {

    int arr[] = null;
    int capacity = 100;
    int top = -1;
    int deleted;

    public Stack() {
        this.arr = new int[capacity];
    }

    public Stack(int capacity) {
        this.capacity = capacity;
        this.arr = new int [capacity];
    }

    public void push(int item) {
        if(isFull()) 
            System.out.println("Stack overflow!\nCan't push " + item + " to stack.");
        else {
            arr[++top] = item;
        }
    }

    public void pop() {
        if(isEmpty()) 
            System.out.println("Stack underflow!\nCan't pop anything.");
        else {
            deleted = arr[top--];
            System.out.println(deleted + " Poped...!");
        }
    }

    public boolean isEmpty() {
        if(top == -1) return true;
        else return false;
    }

    public boolean isFull() {
        if(top == capacity-1) return true;
        else return false; 
    }

    public void print() {
        if(isEmpty()) System.out.println("Nothing to print!");
        else {
            for(int i = top; i >= 0; i--) {
                System.out.print(arr[i] + " ");
            }
            System.out.println();
        }
    }

    public int size() {
        return top+1;
    }

    public int peek() {
        if(isEmpty()) return -1;
        return arr[top];
    }

    public static void main(String[] args) {

        Stack stack = new Stack(10);
        for(int i = 0; i < 10; i++) stack.push(i+10);
        stack.print();
        stack.pop();
        stack.print();
        stack.push(20);
        stack.push(21);
        // Stack stack = new Stack();
        // stack.pop();
        int peeked = stack.peek();
        if(peeked == -1) System.out.println("Nothing to peek!");
        else System.out.println(peeked + " peeked from the top position...!");

        System.out.println("Size of the Satck is : " + stack.size());
        
    }
}

File: Stack/stack.cpp
#include<iostream>

using namespace std;

class Stack {
    public:
        int MAX = 100;
        int top = -1;
        int stack[100];

    bool isEmpty() {
        return top == -1;
    }
    bool isFull() {
        return top == MAX-1;
    }

    void pop() {
        if(isEmpty()) cout << "Stack UnderFlow!" << endl;
        else {
            cout << "Poped : " << stack[top] << endl;
            --top;
        }
    }

    void push(int item) {
        if(isFull()) cout << "Stack Overflow!" << endl;
        else {
            stack[++top] = item;
        }
    }

    void display() {
        cout << "Elements of Stacks : ";
        for(int i = top; i >= 0; i--) cout << stack[i] << " ";
        cout << endl;
    }

    int getSize() {
        return top;
    }
};

int main() {
    Stack stack;

    for(int i = 10; i >= 1; i--) stack.push(i);

    stack.display();

    stack.pop();

    stack.display();

    cout << "Size of Stack : " << stack.getSize() << endl;

    return 0;
}
